lib/shelley/src/Cardano/Wallet/Shelley/Transaction/Size.hs

```diff
@@ -0,0 +1,234 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+--
+-- Size estimation of cbor-encoded transactions in Shelley
+
+module Cardano.Wallet.Shelley.Transaction.Size
+    ( sizeOfTxIn
+    , sizeOfSignedTx
+    , sizeOfTxWitness
+    , sizeOfTxOut
+    , sizeOfCoin
+    , maxSizeOf
+    , MaxSizeOf
+    , minSizeOf
+    , MinSizeOf
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth, NetworkDiscriminant (..) )
+import Cardano.Wallet.Primitive.AddressDerivation.Byron
+    ( ByronKey )
+import Cardano.Wallet.Primitive.AddressDerivation.Icarus
+    ( IcarusKey )
+import Cardano.Wallet.Primitive.Types
+    ( Address (..), Coin (..), TxIn (..), TxOut (..) )
+
+import qualified Codec.CBOR.Encoding as CBOR
+import qualified Codec.CBOR.Write as CBOR
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+
```

Comment:
I assume none of these :point_down: will apply to Shelley on-chain data-types. We should adjust that based on the CDDL spec.

Reply:
Can do when adding tx support, I'd say.

Reply:
Yes, but in the meantime, I'd not add this at all since we know it doesn't apply already.